src/ocean/sys/Stats.d

```diff
@@ -236,3 +240,197 @@ public class CpuMemoryStats
     }
 
 }
+
+/*******************************************************************************
+
+    Class used for getting used GC stats for the current process. This will
+    work only with dmd-transitional.
+
+*******************************************************************************/
+
+public class GCStats
+{
+    static if ( hasMember!(GC, "monitor") )
+    {
+        /**********************************************************************
+
+            The type of the gcEnd delegate
+
+        **********************************************************************/
+
+        private alias ParameterTupleOf!(GC.monitor)[1] MonitorEndParams;
+
+        /**********************************************************************
+
+            Integers used by the gcEnd delegate
+
+        **********************************************************************/
+
+        private alias ParameterTupleOf!(MonitorEndParams)[0] MonitorInt;
+    }
+    else
+    {
+        /**********************************************************************
+
+            Default type for intgers used by monitor
+
+        **********************************************************************/
+
+        private alias long MonitorInt;
+    }
+
+    /**********************************************************************
+
+        Structure representing recorded GC stats.
+
+    **********************************************************************/
+
+    public struct Stats
+    {
+        /**********************************************************************
+
+            The number of microseconds for how much the garbage collector
+            was run.
+
+        **********************************************************************/
+
+        public size_t gc_run_duration;
+
+        /**********************************************************************
+
+            The percentage of time that was spent by the garbage collector.
+
+        **********************************************************************/
+
+        public float gc_run_percentage = 0;
+
+        /**********************************************************************
+
+            The number of bytes freed overall
+
+        **********************************************************************/
+
+        int freed;
+
+        /**********************************************************************
+
+            The number of bytes freed within full pages
```

Comment:
This member could have a better explanation for the users (without them knowing about the GC internals)

Reply:
I'm not an expert in GC internals... any suggestions for how I can reword this?

Reply:
It is actually implementation defined. In dmd-transitional both `free` values will be the same - https://github.com/sociomantic-tsunami/dmd-transitional/blob/v2.078.3.sX/patches/druntime/0004-Add-temproary-GC.monitor-implementation.patch#L127

Thinking about this more, I'd suggest to only use `GC.monitor` only for timing stats. Additional GC stats can optionally be retrieved by manual `GC.stats` call and this will be more future-proof.

Reply:
done